src/rubrix/client/api.py

```diff
@@ -401,26 +400,32 @@ def load(
         query: Optional[str] = None,
         ids: Optional[List[Union[str, int]]] = None,
         limit: Optional[int] = None,
-        as_pandas: bool = True,
-    ) -> Union[pandas.DataFrame, Dataset]:
-        """Loads a dataset as a pandas DataFrame or a Dataset.
+        as_pandas=None,
+    ) -> Dataset:
+        """Loads a Rubrix dataset.
 
         Args:
             name: The dataset name.
             query: An ElasticSearch query with the
                 `query string syntax <https://rubrix.readthedocs.io/en/stable/guides/queries.html>`_
             ids: If provided, load dataset records with given ids.
             limit: The number of records to retrieve.
-            as_pandas: If True, return a pandas DataFrame. If False, return a Dataset.
+            as_pandas: DEPRECATED! To get a pandas DataFrame do ``rb.load(...).to_pandas()``.
 
         Returns:
-            The dataset as a pandas Dataframe or a Dataset.
+            A Rubrix dataset.
 
         Examples:
             >>> import rubrix as rb
-            >>> dataframe = rb.load(name="example-dataset")
             >>> dataset = rb.load(name="example-dataset")
         """
+        if as_pandas is not None:
+            warnings.warn(
+                "The argument `as_pandas` is deprecated and will be removed in a future version. "
+                "Please adapt your code accordingly. "
+                "If you want a pandas DataFrame do `rb.load(...).to_pandas()`."
+            )
+
```

Comment:
This method will work if `as_pandas=False` but the user flow will fail when passing `as_pandas=True`. Also, the error can be confusing for the user.

Maybe, we should include a check here to raise a `ValueError` when `as_pandas=True` 

```suggestion
        if as_pandas is False:
            warnings.warn(
                "The argument `as_pandas` is deprecated and will be removed in a future version. "
                "Please adapt your code accordingly. "
            )
        elif as_pandas is True:
           raise ValueError(
                "If you want a pandas DataFrame do `rb.load(...).to_pandas()`."
            )

```

Reply:
:+1: Done, I just made the error message a bit more explicit.